### ISSUE TITLE
Issue with Furman Pivi model when simulating Stainless Steel

### ISSUE THREAD

**Issue by giadarol** [PyCOMPLETE/PyECLOUD#135]
Fermilab colleagues showed that the code hangs in the initialization stage when measuring the SEY when they configure the Furman Pivi model with stainless steel parameters from https://journals.aps.org/prab/pdf/10.1103/PhysRevSTAB.5.124404

**Comment by giadarol** [comment]
For grazing impact angle, the Poisson generator gives a number of macroparticle much higher than the max specified and it does not manage to get an acceptable number over many retrials

**Comment by giadarol** [comment]
This can be solved by getting the maximum allowed number of generated macroparticles after a few retrials.

**Comment by giadarol** [comment]
Additional inputs have been introduced in PR #136 to limit the angle range inside which the angular dependence is applied. See example below:

```python
furman_pivi_surface = {
'use_modified_sigmaE': False,
'use_ECLOUD_theta0_dependence': False,
'use_ECLOUD_energy': False,
'conserve_energy': False,
'exclude_rediffused': False,
'choice': 'poisson',
'M_cut': 10,
'p_n': np.array([1.6, 2.0, 1.8, 4.7, 1.8, 2.4, 1.8, 1.8, 2.3, 1.8]),
'eps_n': np.array([3.9, 6.2, 13.0, 8.8, 6.25, 2.25, 9.2, 5.3, 17.8, 10.0]),
'p1EInf': 0.07,
'p1Ehat': 0.5,
'eEHat': 0.0,
'w': 100.0,
'p': 0.9,
'e1': 0.26,
'e2': 2.,
'sigmaE': 1.9,
'theta_e_max': 40 * np.pi / 180,
'p1RInf': 0.74,
'eR': 40.,
'r': 1.,
'q': 0.4,
'r1': 0.26,
'r2': 2.,
'theta_r_max': 40 * np.pi / 180,
'deltaTSHat': 1.22000,
'eHat0': 310.0,
's': 1.813,
't1': 0.66,
't2': 0.8,
't3': 0.7,
't4': 1.
}
```

**Comment by giadarol** [comment]
Soved in PR #136 (merged in v8.6.0)
